### PR TITLE
feat: migrate tracing reporter from jaeger agent to opentelemetry collector (otlp)

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Closes https://github.com/datafuselabs/databend/issues/13370
- ENV `DATABEND_JAEGER_ENDPOINT` is replaced by `DATABEND_OTEL_EXPORTER_OTLP_ENDPOINT`.

QuMOwi7R47
9621fa19-f190-4e26-a0ae-e921d466bad7

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13381)

### CODE REVIEW

**Comment by BohuTANG** [comment]
cc @drmingdrmer 

test_impl_raft_storage panic:
```
--- STDOUT:              databend-meta::it store::test_impl_raft_storage ---

running 1 test
test store::test_impl_raft_storage ... FAILED

failures:

failures:
    store::test_impl_raft_storage

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 58 filtered out; finished in 0.02s


--- STDERR:              databend-meta::it store::test_impl_raft_storage ---
thread 'store::test_impl_raft_storage' panicked at 'there is no reactor running, must be called from the context of a Tokio 1.x runtime', /opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/tonic-0.8.3/src/transport/service/executor.rs:15:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by BohuTANG** [comment]
Some of the mysql handler tests is very slow:
image

File: src/common/tracing/src/config.rs [hunk 1]
```diff
@@ -46,7 +46,7 @@ impl Config {
             tracing: TracingConfig {
                 on: true,
```
Comment (zhang2014) [inline src/common/tracing/src/config.rs:47]:
```suggestion
                on: false,
```

Comment (andylokandy) [reply src/common/tracing/src/config.rs:47]: It has been on for a long time, and meta's unittest needs to report traces.

**Comment by zhang2014** [comment]
> Some of the MySQL handler tests are running slowly. Please take a look @zhang2014 

disabled trace for mysql handler unit test in #13389

**Comment by BohuTANG** [comment]
https://github.com/datafuselabs/databend/pull/13391 merged, let's continue 🚀

**Comment by andylokandy** [comment]
The unit test is still slow.

**Comment by andylokandy** [comment]
seems opentelemetry reporter is blocking the only thread

**Comment by BohuTANG** [comment]
> seems opentelemetry reporter is blocking the only thread

Put it in a single thread? The old way is in a spawn thread.

**Review by zhang2014 — APPROVED** [review]

File: src/common/tracing/src/minitrace.rs [hunk 2]
```diff
@@ -76,24 +78,43 @@ pub fn init_logging(name: &str, cfg: &Config) -> Vec<Box<dyn Drop + Send + Sync
...
-                    .build_collector_exporter::<opentelemetry::runtime::Tokio>()
-                    .expect("initialize jaeger exporter"),
-                opentelemetry::trace::SpanKind::Server,
-                Cow::Owned(opentelemetry::sdk::Resource::default()),
-                opentelemetry::InstrumentationLibrary::new(name, None, None),
-            )
+        let otlp_endpoint = cfg.tracing.otlp_endpoint.clone();
+
+        let (reporter_rt, otlp_reporter) = std::thread::spawn(|| {
+            let rt = tokio::runtime::Runtime::new().unwrap();
```
Comment (zhang2014) [inline src/common/tracing/src/minitrace.rs:84]:
```suggestion
            let rt = tokio::runtime::Runtime::new().worker_threads(2).unwrap();
```

Comment (andylokandy) [reply src/common/tracing/src/minitrace.rs:84]: This runtime is only for opentelemetry and it seems only use and block one thread.

Comment (BohuTANG) [reply src/common/tracing/src/minitrace.rs:84]: I will modify in anthor PR later, this settting will make the reporter only can use `2 threads `.

Comment (BohuTANG) [reply src/common/tracing/src/minitrace.rs:84]: Fix in:
https://github.com/datafuselabs/databend/pull/13402/files#diff-0cd0738e41abd997179229a48eaed578ee7af4bc2103df9c3e1c743e76a18a6bR85

**Pull request merged**